controls/radmap/how-to/howto-wraparound-virtualization-visualizationlayer.md

```diff
@@ -0,0 +1,383 @@
+---
+title: Wraparound with Virtualization in VisualizationLayer
+page_title: Wraparound with Virtualization in VisualizationLayer
+description: Wraparound with Virtualization in VisualizationLayer
+slug: radmap-howto-wraparound-virtualization-visualizationlayer
+tags: wraparound,virtualization,visualizationlayer,map
+published: True
+position: 10
+---
+
+# Wraparound with Virtualization in VisualizationLayer
+
+When using wraparound functionality, the virtualization of the RadMap control needs to be set in a more advanced way. 
+
+![Wraparound with Virtualization in VisualizationLayer](images/RadMap_Wraparound_Virtualization_Visualizationlayer.PNG)	
+
+In order to enable the UI Virtualization feature, you should configure the ZoomLevelGridList object and you should specify the VirtualizationSource property of the [Visualization Layer]({%slug radmap-features-virtualization-layer%}). The ZoomLevelGridList property allows dividing the map into regions. The data provided to the VisualizationLayer should be wrapped in a class that implements the __IMapItemsVirtualizationSource__ interface. This interface contains a __MapItemsRequest()__ method, which is used by the __VisualizationLayer__ to request new data whenever the ZoomLevel or Center properties of the RadMap control are changed. The event arguments are of type __MapItemsRequestEventArgs__ which give you the currently traversed portion of the map. This method generally serves as an event handler and you have to calculate and decide which objects to be displayed in the given portion. In WrapAround mode, this portion will have Longitude value exceeding the valid geographical boundaries (-180; 180). The RadMap control provides public methods which you can use to normalize such locations. You can take a look a the [Location Conversion]({%slug radmap-features-wraparound%}) section in the WrapAround help article.
```

Comment:
"which give you the currently traversed portion of the map."

I am more or less familiar with the map, but I don't know what is "traversed portion". Lets use a phrase that is easier to understand. Maybe, something like "the visible area of the map" or "the viewport".